src/wallet/feebumper.cpp

```diff
@@ -245,6 +245,19 @@ Result CreateRateBumpTransaction(CWallet& wallet, const uint256& txid, const CCo
         }
     }
 
+    // If no recipients, means that we are sending coins to a change address
+    if (recipients.empty()) {
+        // Just as a sanity check, ensure that the change address exist
+        if (std::get_if<CNoDestination>(&new_coin_control.destChange)) {
+            errors.emplace_back(Untranslated("Unable to create transaction. Transaction must have at least one recipient"));
+            return Result::INVALID_PARAMETER;
+        }
+
+        // Add change as recipient with SFFO flag enabled, so fees are deduced from it and no other output is created.
+        recipients.emplace_back(CRecipient{GetScriptForDestination(new_coin_control.destChange), output_value, /*fSubtractFeeFromAmount=*/true});
+        new_coin_control.destChange = CNoDestination();
+    }
```

Comment:
In 7cb4dc157a711210e18f3c9b492150b6eb984b30 " bumpfee: enable send coins back to yourself " 

While this certainly looks correct, I think there is more concise way of doing this within the outputs loop before this one. I've implemented it here if you would like to take a look: https://github.com/ishaanam/bitcoin/commit/5b1017c1ea11c8fe2351a9110e4c6ddad6e77f8a. The advantage of this would be that the sanity check would not be required and imo it would be a bit more readable.

Reply:
Thanks ishaanam.
At least for me, it doesn't look right to check the vector size while we loop over it. It is also not correct to use `outputs.size()` as `outputs` will be empty if the user didn't customize them.

Reply:
> It is also not correct to use outputs.size() as outputs will be empty if the user didn't customize them.

In what scenario would we need to make the change output the sole recipient if the user didn't customize the outputs? My understanding was that this is only an issue because of the addition of the "outputs" vector.

Reply:
a send-to-self, single change output tx that is bumped.